src/Text/Pandoc/Readers/LaTeX.hs

```diff
@@ -2362,21 +2361,63 @@ parseTableRow envname prefsufs = do
   cells <- mapM (\ts -> setInput ts >> parseTableCell) rawcells
   setInput oldInput
   spaces
-  let numcells = length cells
-  guard $ numcells <= cols && numcells >= 1
-  guard $ cells /= [mempty]
-  -- note:  a & b in a three-column table leaves an empty 3rd cell:
-  return $ cells ++ replicate (cols - numcells) mempty
+  return $ Row nullAttr cells 
 
-parseTableCell :: PandocMonad m => LP m Blocks
+parseTableCell :: PandocMonad m => LP m Cell
 parseTableCell = do
-  let plainify bs = case toList bs of
-                         [Para ils] -> plain (fromList ils)
-                         _          -> bs
   updateState $ \st -> st{ sInTableCell = True }
-  cells <- plainify <$> blocks
+  cell' <- parseMultiCell <|> parseSimpleCell
   updateState $ \st -> st{ sInTableCell = False }
-  return cells
+  return cell'
+
+cellAlignment :: PandocMonad m => LP m Alignment
+cellAlignment = skipMany (symbol '|') *> (left <|> right <|> center <|> def') <* skipMany (symbol '|')
+  where
+    left   = AlignLeft    <$ char' 'l'
+    right  = AlignRight   <$ char' 'r'
+    center = AlignCenter  <$ char' 'c'
+    def'   = AlignDefault <$ symbol '*'
+
+    char' c = satisfyTok isc >> pure c
+      where isc (Tok _ Word d) = maybe False (\(c', _) -> c == c') (T.uncons d)
+            isc _ = False
+
+plainify :: Blocks -> Blocks
+plainify bs = case toList bs of
+                [Para ils] -> plain (fromList ils)
+                _          -> bs
+
+parseMultiCell :: PandocMonad m => LP m Cell
+parseMultiCell =   (controlSeq "multirow"    >> parseMultirowCell) 
+               <|> (controlSeq "multicolumn" >> parseMulticolCell)
+  where
+    parseMultirowCell = parseMultiXCell RowSpan (const $ ColSpan 1)
+    parseMulticolCell = parseMultiXCell (const $ RowSpan 1) ColSpan
+
+    parseMultiXCell rowspanf colspanf = do
+      span' <- untokenize <$> braced
+      let span'' = maybe 1 id $ safeRead span'
+      alignment <- symbol '{' *> cellAlignment <* symbol '}'
+      -- Two possible contents: either a nested \multirow/\multicol, or content.
+      -- E.g. \multirow{1}{c}{\multicol{1}{c}{content}}
+      symbol '{' 
```

Comment:
A line like

```haskell
symbol '{' *> (multiCell <|> singleCell) <* symbol '}'
```

could be a bit cleaner, but would require additional helper functions.